src/python/pants/option/options_bootstrapper_test.py

```diff
@@ -193,7 +193,7 @@ def assert_config_read_correctly(
                 dedent(
                     """\
                     [compile_apt]
-                    worker_count = 1
+                    worker_count = "1"
```

Comment:
It is in the sense that we did unfortunately allow sloppiness thanks to Python runtime type casting, and we can no longer allow it. However we never documented that this would work, and this is the sort of thing we will issue warnings about during the transition. Fortunately the fix should be easy and we can give the answer in the warning.